extended/src/main/java/io/kubernetes/client/extended/leaderelection/LeaderElector.java

```diff
@@ -15,6 +15,41 @@
 
   private static final double JITTER_FACTOR = 1.2;
 
+  public static LeaderElector newLeaderElector(LeaderElectionConfig config) {
+    if (config == null) {
+      throw new RuntimeException("Config must be provided.");
+    }
+    if (config.getLock() == null) {
+      throw new RuntimeException("Lock must be provided.");
+    }
+    if (config.getLeaseDuration() == null) {
+      throw new RuntimeException("LeaseDuration must be provided.");
+    }
+    if (config.getRetryPeriod() == null) {
+      throw new RuntimeException("RetryPeriod must be provided.");
+    }
+    if (config.getRenewDeadline() == null) {
+      throw new RuntimeException("RenewDeadline must be provided.");
+    }
+    if (config.getLeaseDuration().compareTo(config.getRenewDeadline()) <= 0) {
+      throw new RuntimeException("LeaseDuration must be greater than renewDeadline.");
+    }
+    if (config.getRenewDeadline().compareTo(config.getRetryPeriod()) <= 0) {
+      throw new RuntimeException("RenewDeadline must be greater than retryPeriod.");
+    }
+    if (config.getLeaseDuration().isZero() || config.getLeaseDuration().isNegative()) {
+      throw new RuntimeException("LeaseDuration must be greater than zero.");
+    }
+    if (config.getRenewDeadline().isZero() || config.getRenewDeadline().isNegative()) {
+      throw new RuntimeException("RenewDeadline must be greater than zero.");
+    }
+    if (config.getRetryPeriod().isZero() || config.getRetryPeriod().isNegative()) {
+      throw new RuntimeException("RetryPeriod must be greater than zero.");
```

Comment:
Rather than throwing a bunch of exceptions linearly, it's much better to accumulate all of the errors in a single list and then throw one exception. Something like:

```java
List<String> errors = new ArrayList();
if (config.getRetryPeriod().isZero()) {
   errors.add("retry can't be zero");
}
.... // other if's here.

if (errors.length() > 0) {
   throw new IllegalArgumentException(errors.join(","));
}
```

That way the user has a much better experience rather than fixing an error, re-running, getting a new error, etc etc.